addons/stock/views/stock_move_line_views.xml

```diff
@@ -8,17 +8,20 @@
                 <field name="location_usage" column_invisible="True"/>
                 <field name="location_dest_usage" column_invisible="True"/>
                 <field name="date"/>
-                <field name="reference" string="Reference"
-                       column_invisible="context.get('no_reference', False)"/>
+                <field name="reference" string="Reference" class="fw-bolder"
+                       column_invisible="context.get('no_reference', False)"
+                       decoration-info="picking_code == 'outgoing'"
+                       decoration-success="picking_code == 'incoming'"
+                       decoration-warning="picking_code not in ('incoming', 'outgoing')"/>
                 <field name="product_id"/>
                 <field name="lot_id" optional="show" groups="stock.group_production_lot"/>
                 <field name="package_id" optional="hide" groups="stock.group_tracking_lot"/>
                 <field name="result_package_id" optional="hide" groups="stock.group_tracking_lot"/>
-                <field name="location_id"/>
-                <field name="location_dest_id"/>
+                <field name="location_id" decoration-muted="location_usage != 'internal'"/>
+                <field name="location_dest_id" decoration-muted="location_dest_usage != 'internal'"/>
```

Comment:
should maybe be checked if it should include `transit` locations

Reply:
@lugr-odoo I agree with tic